cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/connector/resource/AwsCloudWatchService.java

```diff
@@ -74,35 +75,51 @@ public void addCloudWatchAlarmsForSystemFailures(List<CloudResource> instances,
         });
     }
 
-    public void deleteCloudWatchAlarmsForSystemFailures(CloudStack stack, String regionName, AwsCredentialView credentialView) {
+    public void deleteAllCloudWatchAlarmsForSystemFailures(CloudStack stack, String regionName, AwsCredentialView credentialView) {
         List<String> instanceIds = stack.getGroups().stream()
                 .flatMap(group -> group.getInstances().stream())
                 .map(CloudInstance::getInstanceId)
                 .collect(Collectors.toList());
+        LOGGER.info("Found ids for running instances: [{}]. Cloudwatch alarms for these instances will be deleted.", instanceIds);
+        List<String> deletedInstanceIds = stack.getGroups().stream()
+            .flatMap(group -> group.getDeletedInstances().stream())
+            .map(CloudInstance::getInstanceId)
+            .collect(Collectors.toList());
+        LOGGER.info("Found ids for deleted instances: [{}]. Deletion will be attempted on the cloudwatch alarms for these instances.", deletedInstanceIds);
+        instanceIds.addAll(deletedInstanceIds);
         deleteCloudWatchAlarmsForSystemFailures(stack, regionName, credentialView, instanceIds);
     }
 
     public void deleteCloudWatchAlarmsForSystemFailures(CloudStack stack, String regionName, AwsCredentialView credentialView, List<String> instanceIds) {
```

Comment:
**question:** Do we need to call this method in `AwsDownscaleService`? Could we use the `deleteAllCloudWatchAlarmsForSystemFailures` method instead?

I think the naming of this method and the one above is a bit confusing, and thought we could just make this one `private` and run everything through the delete _all_.

Would we want to keep cloudwatch alarms around for downscaled instances?

Reply:
Ah, I see you've already made one method private, so presumably you already looked into this?

Reply:
I didn't change the downscale path. I also didn't make any method private that wasn't private already. I renamed `deleteAllCloudWatchAlarmsForSystemFailures` to include the `All` because that method is only used when we're terminating the instance, and so should be deleting all cloudwatch alarms associated with it. That's different than the downscale case, where we're deleting the cloudwatch alarms for the downscaled instances, but want to leave them for the other instances. So calling `deleteAllCloudWatchAlarmsForSystemFailures` wouldn't make sense in that case, because we don't want to delete all.